### PR TITLE
Voeg naming-conventions toe aan standaard

### PR DESCRIPTION
**Pull request by TimvdLippe**
De naming-conventions zijn ontwikkeld als onderdeel van
het kennisplatform APIs als aparte module. Deze commit
integreert de conventions in de standaard als rules
om overhead te voorkomen en alle rules verplicht te
maken (in tegenstelling tot een module die niet altijd
nodig is).

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Preview is te zien op https://logius-standaarden.github.io/Publicatie-Preview/API-Design-Rules/pr/timvdlippe/naming-conventions/

**Comment by TimvdLippe** [comment]
@bwbroersma FYI dit is de wijziging wat betreft naming conventions

**Comment by bwbroersma** [comment]
Is this for the current version 2.0.x that is in procedure at the Netherlands Standardisation Forum, or a later 2.1.x version?

BTW `^[a-z0-9]+([A-Z][a-z0-9]+)*$` does allow single letter first parts, but not single letter in later (uppercased) parts, so `aLabel` is allowed, while `labelA` is not. Maybe only a lower case first letter can be enforced, so e.g. `^[a-z0-9]+[A-Za-z0-9]*$`.

**Comment by TimvdLippe** [comment]
@bwbroersma This is for a future version of ADR, unrelated to the ongoing 2.0.x procedure.

Yes good point. I wanted to be clear about multiple consecutive uppercase characters which aren't allowed in camelCase. I updated the regex to handle that edge case. You can test it out here: https://regex101.com/r/7nmJmc/2

**Comment by TimvdLippe** [comment]
De huidige linter (toegevoegd in #174) vereist kebab-case al voor paths. Voor query keys kunnen we nog een validatie schrijven. De afkortingen regel is niet te automatiseren helaas, maar zou manueel moeten worden gecheckt.

**Review by sanderke — APPROVED** [review]

**Comment by erwinkramer** [comment]
> Voor query keys kunnen we nog een validatie schrijven. 

Wat denk je van https://github.com/erwinkramer/bank-api/blob/f7f5362a994ed9b4eb1ad2650873c385cb6a5ee4/.spectral/ruleset.bank.yml#L45 ?

**Comment by TimvdLippe** [comment]
We hebben meer issues binnen gekregen wat betreft naming conventions. We vergroten daarom de scope van deze wijziging om een groter discussiepunt te behandelen wat betreft naming van verscheidene onderdelen.

Vanuit beheer zien we naming conventions zeker als nuttig. Echter willen we niet alle wijzigingen nu al voorbereiden als later blijkt dat het TO anders beslist. Daarom gebruiken we deze wijziging als basis voor de discussie.

**Comment by TimvdLippe** [comment]
Ik heb de merge conflicts opgelost, de linter referentie toegevoegd en styling voor de correct/incorrect voorbeelden.

File: sections/designRules.md [hunk 1]
```diff
@@ -103,6 +137,88 @@ A resource describing a single thing is called a [=singular resource=]. Resource
...
+            <pre class="nohighlight example-correct">https://api.example.org/v1/organisaties/_zoek</pre>
+            <p>URI path included in <code>openapi.json</code> with camelCase variable (correct):</p>
+            <pre class="nohighlight example-correct">https://api.example.org/v1/organisaties/{organisatieId}</pre>
+         </div>
+      </dd>
+      <dt>How to test</dt>
+      <dd>
+         Loop all resource paths in the OpenAPI Description and check that all resource path segments use lowercase letters, digits or hyphens (<code>-</code>). The last path segment is allowed to start with a <code>_</code>. You can use the following regex for each resource path:
+         <div class="example">
+            <pre><code>^(\/|(\/_[a-z]+|\/(([a-z\-]+|{[a-z]+})(\/([a-z\-\.]+|{[a-z]+}))*)(\/_[a-z]+)?))$</code></pre>
```
Comment (sanderke) [inline sections/designRules.md:186]:
```suggestion
            <pre><code>^(\/|(\/_[a-z0-9]+|\/(([a-z0-9\-]+|{[^}]+})(\/([a-z0-9\-\.]+|{[^}]+}))*)(\/_[a-z]+)?)\/?)$</code></pre>
```
Hier ook kebab met digits?

**Review by sanderke — APPROVED** [review]

**Comment by TimvdLippe** [comment]
Vandaag besproken in het TO. De volgende opmerkingen/conclusies zijn geplaatst:

- Positief ten opzichte van standaardiseren van naming conventions
- Afkortingen is subjectief, niet te automatiseren en soms juist handig
- Besloten om de regel met afkortingen niet te standaardiseren
- Path segments kebab-case en query keys camelCase zijn in principe goed
- Moet wel gekeken worden naar impact op informatiemodellen
- Ook moeten nog wat meer details worden aangepakt op basis van bestaande API's (zoals ZGW)

De afkortingen regel verwijderen we. De andere regels moeten nog verder worden uitgewerkt in aparte dedicated sessies. Bij een volgend TO zal het weer besproken worden.

File: sections/designRules.md [hunk 2]
```diff
@@ -103,6 +103,84 @@ A resource describing a single thing is called a [=singular resource=]. Resource
...
 
+<div class="rule" id="/core/path-segments-kebab-case" data-type="technical">
+   <p class="rulelab">Use kebab-case in path segments</p>
+   <dl>
+      <dt>Statement</dt>
+      <dd>
+         <div>
+            <p>Path segments of a [=URI=] MUST only contain lowercase letters, digits or hyphens. This is also known as <a href="https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case">kebab-case</a>. Hyphens MUST only be used to deliniate distinct words. This also implies that diacritics MUST be normalized and special characters MUST be omitted.
+            <p>Another implication of this rule is that file extensions MUST NOT be used. Resources SHOULD use the <code>Accept</code> header for content negotation.
+            <p>The last path segment MAY start with `_`, which is used as a convention to implement search API's
```
Comment (sanderke) [inline sections/designRules.md:114]: Uit overleg 4 juni: hier verwezen naar `/core/resource-operations`

Comment (sanderke) [inline sections/designRules.md:114]: Uit overleg 4 juni: hier verwezen naar `/core/resource-operations`

File: sections/designRules.md [hunk 3]
```diff
@@ -103,6 +103,84 @@ A resource that corresponds to a single conceptual entity is referred to as a [=
...
+   </dl>
+</div>
+
+<div class="rule" id="/core/query-keys-camel-case" data-type="technical">
+   <p class="rulelab">Use camelCase in query keys</p>
+   <dl>
+      <dt>Statement</dt>
+      <dd>
+         <div>
+            <p>Query keys in a [=URI=] MUST only contain letters and digits, where the first letter of each word is capitalized, except for the first letter of the entire compound word. This is also known as <a href="https://developer.mozilla.org/en-US/docs/Glossary/Camel_case">lower camelCase</a>. This also implies that diacritics MUST be normalized and special characters MUST be omitted.
```
Comment (joostfarla) [inline sections/designRules.md:161]: Het lijkt me goed om te vereisen dat een query key altijd start met een letter (geen cijfer).

**Comment by TimvdLippe** [comment]
Deze wijziging is geaccepteerd in het TO van 2025-09-02.
